src/modules/addimage.js

```diff
@@ -734,28 +734,15 @@ import { atob, btoa } from "../libs/AtobBtoa.js";
   var arrayBufferToBinaryString = (jsPDFAPI.__addimage__.arrayBufferToBinaryString = function(
     buffer
   ) {
-    // Skip direct conversion as it might take many hundred milliseconds before throwing.
-    if (buffer.length < ARRAY_BUFFER_LIMIT) {
-      try {
-        return atob(btoa(String.fromCharCode.apply(null, buffer)));
-      } catch (e) {
-        // Buffer was overflown, fall back to other methods
-      }
-    }
-    // Text decoder is much faster than string concatenation or reduce/join.
-    if (typeof TextDecoder !== "undefined") {
-      var decoder = new TextDecoder("ascii");
-      return decoder.decode(buffer);
-    } else if (
-      typeof Uint8Array !== "undefined" &&
-      typeof Uint8Array.prototype.reduce !== "undefined"
-    ) {
-      return new Uint8Array(buffer)
-        .reduce(function(data, byte) {
-          return data.push(String.fromCharCode(byte)), data;
-        }, [])
-        .join("");
-    }
+    var out = "";
+    var u8buf = new Uint8Array(buffer);
```

Comment:
I actually updated the above jsbench suite to include a test without creating the extra Uint8Array and that creating it is 16% slower than direct buffer access: https://jsbench.me/x5kmnswnmz/1

Was there a reason for adding this @101arrowz ?

Reply:
Which tests specifically are you comparing? The only one that doesn't convert to `Uint8Array` ("Direct apply") fails with "Maximum stack size exceeded". Besides, you absolutely need to convert to `Uint8Array` because you cannot access indices on an `ArrayBuffer`. Try this in the console:

```js
console.log(new ArrayBuffer(1)[0]); // undefined
console.log(new Uint8Array(1)[0]); // 0
```

Reply:
Oh yeah, in my case the function input is Uint8Array already but that is called with buffers as well!
I meant comparing the last vs 2nd

Reply:
I don't follow; does the jsPDF version accept only `ArrayBuffer`, only `Uint8Array`, or both? If only `ArrayBuffer`, use what I wrote; if `Uint8Arary`, use what you wrote; if both, check if it's an ArrayBuffer via `instanceof` and if so, convert to `Uint8Array` before use.

Reply:
Yeah, checked the calls and it's kind of 50/50 called with `ArrayBuffer` and `Uint8Array`.